python/kfserving/kfserving/test_storage.py

```diff
@@ -34,3 +37,31 @@ def test_no_prefix_local_path():
     relative_path = '.'
     assert kfserving.Storage.download(abs_path) == abs_path
     assert kfserving.Storage.download(relative_path) == relative_path
+
+
+def test_public_gcs():
+    gcs_path = 'gs://kfserving-samples/models/tensorflow/flowers'
+    assert kfserving.Storage.download(gcs_path)
```

Comment:
It's typically bad practice for unit tests to reach out into the real world like this. Can we mock the minio and gcs client instead? Otherwise we're implementing non-hermetic which can be extremely brittle to changes -- like if I delete this file.

Reply:
I think we can use python `MagicMock` to mock the response and test different exceptions.
https://docs.python.org/3/library/unittest.mock.html

Reply:
In my recent commits, I used magicmock and mock.patch to simulate some of the outputs for unit tests.